sbt-play-ebean/src/main/scala/play/ebean/sbt/PlayEbean.scala

```diff
@@ -25,7 +40,24 @@ object PlayEbean extends AutoPlugin {
 
   override def trigger = noTrigger
 
-  override def projectSettings = inConfig(Compile)(scopedSettings) ++ unscopedSettings
+  override def projectSettings = {
+    val querySettings = Seq(
+      playEbeanQueryGenerate := false,
+      playEbeanQueryEnhance := false,
+      playEbeanQueryDestDirectory := "app",
```

Comment:
I think it needs to go into a managed source directory, assuming it's a Java file. There's actually a better way to configure directories. Take a look at the Twirl sbt plugin to see how it does it:
https://github.com/playframework/twirl/blob/master/sbt-twirl/src/main/scala/play/twirl/sbt/SbtTwirl.scala.

Concretely here are some steps:
1. Create a new task called `playEbeanQuery`. Call this task from `ebeanEnhance` if `playEbeanQueryGenerate` is created.
2. The task should read from `sources in playEbeanQuery` and output to `target in playEbeanQuery`.
2. Add `playEbeanQuery` task configuration into `scopedSettings` and define values like `sources in playEbeanQuery := resourceDirectory.value` and `target in playEbeanQuery := crossTarget.value / "ebean-query" / Defaults.nameForSrc(configuration.value.name)`.
3. Bind these settings using `inConfig(Compile)(scopedSettings) ++ inConfig(Test) ++ ...` so it will pick up the different directory values when compiling and testing (see [raw and configured settings](http://www.scala-sbt.org/0.13/docs/Plugins-Best-Practices.html#Provide+raw+settings+and+configured+settings)).

* Here is the best practices guide which suggests reusing keys: http://www.scala-sbt.org/0.13/docs/Plugins-Best-Practices.html#Reuse+existing+keys.
* Here is the list of keys: http://www.scala-sbt.org/0.13/sxr/sbt/Keys.scala.html.
* You can see how Play binds keys too, so you can reference directories like the `app` directory without hardcoding them: https://github.com/playframework/playframework/blob/master/framework/src/sbt-plugin/src/main/scala/play/sbt/PlayLayoutPlugin.scala.